### PR TITLE
Missing MerchantIdentifier in header using new FeedGen.

### PR DESCRIPTION
**Pull request by BarryCarlyon**
Basically a clone of #75 but for the Ack. data instead.

Added the associated test.

Probably could so with adding a test for dealing with item not being provided as it's optional. But Test coverage isn't my strong point

### CODE REVIEW

File: test/xml.js [hunk 1]
```diff
@@ -36,6 +36,33 @@ describe('xml', () => {
 
                 expect(FeedContent).to.equal(xml);
             });
+
+            it('should handle a multiple orders', () => {
```
Comment (dciccale) [inline test/xml.js]: shouldn't this assertion be "should handle multiple orders" instead?

**Comment by dciccale** [comment]
LGTM so far 👍

File: test/xml.js (hunk 1 above)
Comment (BarryCarlyon) [reply test/xml.js]: I preserved the grammar error from `it('should handle a multiple fulfillments', () => {`

**Comment by BarryCarlyon** [comment]
As per notes in 75, feeds are being rejected by amazon due to no ability for the module to create a valid header with a `MerchantIdentifer`

File: test/xml.js (hunk 1 above)
Comment (dciccale) [reply test/xml.js]: Could you fix that grammar error as well? :)

**Comment by dciccale** [comment]
Hm.. I'll see how it was done before, since I am using the previous version still. And it is working correctly. Maybe I could bring this as a fix

**Comment by BarryCarlyon** [comment]
Follow up in #83

**Comment by BarryCarlyon** [comment]
I suppose this makes it a breaking change?

Version bump needed?

**Comment by dciccale** [comment]
Actually, the current npm version 2.1.0 was not broken. And 2.2.0 (current package.json on github) was not released yet to npm, so we could just roll out 2.2.0 with all these fixes as a minor version since there are no actual breaking changes from 2.1.0.

does make sense?

**Comment by BarryCarlyon** [comment]
Indeed yes.

I wondered why I got 2.1 from NPM last time when 2.2 was here

**Comment by dciccale** [comment]
Ok I was wrong on my previous comment, actually it is a breaking change. Not for the output part, but for the input part of the feeds where now the header must be passed.

I correct myself, this should go to 3.0 directly due to this change.

@BarryCarlyon  Before merging, aren't we breaking other Feeds where the new header argument should be passed in?

@tyscorp @aaronbean just to keep you in the loop, changes Introduced by me on the multi message feature had some bugs (sorry) now are being fixed on this PR + Order ack being multi messages as well. For making multi messages work properly on Amazon we had to introduce a breaking change by adding MerchantIdentifier to a header object being passed to the Feed generator function, to separate this from the <Messages> part of the output (before it was all together).

**Comment by BarryCarlyon** [comment]
Technically no. We are not breaking any other feeds as ALL the other feeds that are defined are just throwing back a `errorNotImplemented(FEEDNAME)` the only Feeds with a "object maker", in the four files in lib/feeds are orders.js with `_POST_ORDER_ACKNOWLEDGEMENT_DATA_` and `_POST_ORDER_FULFILLMENT_DATA_`

So, breaking change **yes**, breaking other feeds **no**, because none of them are implemented.

**Comment by dciccale** [comment]
OK then this should be good to merge.

Let's see if some of the authors have any more comments on this.

If we're going to 3.0.0, I have a branch with ESLint implemented based on airbnb styleguide. Which should help us improve a bit potential errors and having a normalized style. That we could ship on this major also.

**Comment by BarryCarlyon** [comment]
In other news, jsut setup my first orderfulfillment attempt. And discovered that the definition is lacking some keys so patching and testing, extending this PR again tomorrow after some more testing!

_POST_ORDER_FULFILLMENT_DATA_ lackings keys under item and item should be able to support more than one item.

**Comment by dciccale** [comment]
Good catch. However Items is optional here. And should be optional in the
library too. I am currently using this in production without sending items.

**Comment by BarryCarlyon** [comment]
Yeah the XSD https://images-na.ssl-images-amazon.com/images/G/01/rainier/help/xsd/release_1_9/OrderFulfillment.xsd has Item set to minOccurs 0 max infinity.

I'll revise the PR

**Comment by BarryCarlyon** [comment]
The map function should handle it happily the same as `_POST_ORDER_ACKNOWLEDGEMENT_DATA_` and my feed has now been accepted by amazon.

**Comment by BarryCarlyon** [comment]
Oh I didn't patch the tests…

I wonder if we should consider adding some XSD validation.

EG, you can specify one of `AmazonOrderID` or `MerchantOrderID` in `OrderFulfillment` but not both?

**Comment by dciccale** [comment]
Yep tests would be good.

Reg. XSD validation, seems a good idea however it could be rather complex to implement. Plus we should be confident relying on the validation amazon is doing when posting the generated file, and error responses are descriptive enough I think to know how to fix the document.

**Comment by dciccale** [comment]
@BarryCarlyon could you bump package.json to 3.0.0? Is there anything else to be added on this PR? It looks good from my side

**Comment by BarryCarlyon** [comment]
I'm gonna throw a configuration in for refunds (thats this weeks $dayjob task)

Also could do with #66 being considered?

Why are we forcing binary which trashes foreign chars?

**Comment by dciccale** [comment]
That we should discuss on the designated issue / pr.

Reg. this PR I assume it is good to go, will merge it later today if no more comments

**Comment by BarryCarlyon** [comment]
Sounds good. I'll bump the version real quick then

**Comment by dciccale** [comment]
Awesome job @BarryCarlyon thanks

**Pull request merged**